src/components/PopupWithForm.js

```diff
@@ -22,9 +23,31 @@ export default class PopupWithForm extends Popup {
     this._modalForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
       const inputValues = this._getInputValues();
-      console.log("Input values before submit:", inputValues);
+      this.renderSaving(true);
       this._handleFormSubmit(inputValues);
+      this.renderSaving(false);
     });
     super.setEventListeners();
   }
+
+  stopSaving(popup) {
+    popup._modalForm.querySelector(".modal__save-button").textContent =
+      popup._buttonText;
+    console.log("Found _modalForm and done");
+    popup.close();
+  }
+
+  renderSaving(isSaving) {
+    if (isSaving) {
+      this._modalForm.querySelector(".modal__save-button").textContent =
```

Comment:
**Could be improved**

All elements found **every time** you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `getView`) only **1 time** in order not to waste resources on searching them again and again.

`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form`, `this._likeButton`, and so on.